tests/src/resultcollection.hh

```diff
@@ -0,0 +1,170 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+#include "testcommon.hh"
+
+#include <ikarus/finiteelements/ferequirements.hh>
+
+template <typename FiniteElement>
+auto getVertexPositions(FiniteElement& fe) {
+  constexpr int dim            = FiniteElement::Traits::mydim;
+  const auto& element          = fe.gridElement();
+  const auto& referenceElement = Dune::referenceElement<double, dim>(element.type());
+  const int nodes              = referenceElement.size(dim);
+
+  std::vector<typename FiniteElement::GridElementEntityType::Geometry::LocalCoordinate> positions;
+  for (auto i : std::views::iota(0, nodes))
+    positions.push_back(referenceElement.position(i, dim));
+
+  return positions;
+}
+
+inline auto linearStressResultsOfSquare = [](auto& nonLinearOperator, auto& fe) {
+  constexpr int nodes      = 4;
+  constexpr int quantities = 3;
+
+  Eigen::Matrix<double, nodes, quantities> expectedStress;
+  if constexpr (requires { fe.setEASType(4); }) {
+    fe.setEASType(4);
+    expectedStress << 1214.28571429, 1214.28571429, 384.61538462, 1214.28571429, 214.28571429, 384.61538462,
+        214.28571429, 1214.28571429, 384.61538462, 214.28571429, 214.28571429, 384.61538462;
+  } else {
+    expectedStress << 1428.57142857, 1428.57142857, 769.23076923, 1098.90109890, 329.67032967, 384.61538462,
```

Comment:
I think since it is a single element test, the top right corner sees it as a rigid body translation (see deformed configuration in blue below) and shows zero stresses. I am assuming it should give better results with mesh refinement and thereby the solution from EAS is more correct. Does this make sense? This has to be investigated further though. The solution for a single element without EAS for shear strain for example is $2-\xi-\eta$ whereas EAS4 adds additional strains of $-1+\xi+\eta$ making the shear strain as $1$ in the whole element.
Also, the stress values written here only have 10 digits after the decimal point. Do you think we should add more digits and check it against a tolerance of 1e-15?

![image](https://github.com/ikarus-project/ikarus/assets/92575489/f8991805-bfed-4dac-a0b7-21617e701da3)